garbage.c

```diff
@@ -46,8 +46,8 @@ int domain_fork_particle(int parent) {
         ActiveParticle[childactive] = child;
     }
 
-    P[parent].Generation ++;
-    uint64_t g = P[parent].Generation;
+    SPHP(parent).Generation++;
```

Comment:
This will crash if parent is not SPH.

Reply:
By keeping Generation in P forking doesn't need to be aware of particle type. I think we may be able to merge generation with ID, which could have automatically done the ID modification.